src/UseEffect/AntiPatterns/Examples/ApiRequest.tsx

```diff
@@ -7,14 +6,21 @@ interface Props {
 
 export function ApiRequestAntiPattern(props: Props) {
   const { firstName } = props;
-  const [localFirstName, setLocalFirstName] = useState(firstName);
-  const { data, isLoading } = useQuery<string>(firstName);
+  const {
+    data: localFirstName,
+    isLoading,
+    error,
+  } = useQuery<string>(firstName, {
+    onError: () => {
+      // show error toast
+    },
+    // onSuccess...
+  });
 
-  useEffect(() => {
-    if (data) {
-      setLocalFirstName(data);
-    }
-  }, [data]);
-
-  return <p>{isLoading ? 'Loading...' : localFirstName}</p>;
+  return (
+    <>
+      <p>{isLoading ? 'Loading...' : localFirstName}</p>
+      {error && <p>Error {error}</p>}
```

Comment:
Promises can fail - make sure you handle and present that information here, or in a toast.